### PR TITLE
Gauresh first PR (Lot of issues to fix and points for others to learn)

### PR DESCRIPTION
**Pull request by pissall20**

### CODE REVIEW

File: FI-Fully Insured/Dental-FI/Cigna-FI/claims-FI/code.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        elif "financialforce" in read_path:
+         clean_fi_dental_cigna_claims_financialforce(read_path, save_path)  
+        elif "saks dental claims" in read_path:
+         clean_fi_dental_cigna_claims_saks(read_path, save_path)
+        elif "HBC" in read_path:
+         clean_fi_dental_cigna_claims_saks_hbc(read_path, save_path)
+        elif "fifth" in read_path:
+         clean_fi_dental_cigna_claims_saks_5th(read_path, save_path) 
+        elif "shopify" in read_path:
+         clean_fi_dental_cigna_claims_shopify(read_path, save_path)
```
Comment (pissall20) [inline FI-Fully Insured/Dental-FI/Cigna-FI/claims-FI/code.py:126]: There are many indentation problems with this piece of code. Please use Pycharm code formatter to format your file before pushing it to github. 

Every folder has similar reports grouped, so I am pretty sure 1 function or max 2 functions should be able to transform all of them, these many functions are not required at all. Please fix your logic and correct this.

File: FI-Fully Insured/Dental-FI/Cigna-FI/enrollment-FI/code.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+def clean_fi_dental_cigna_enrol(file_path, save_file_path):
+    df = pd.read_excel(file_path,
+                       sheet_name="Enrollment by Tier-1",
+                       skiprows=4,
+                       skipfooter=5,
+                       engine="openpyxl"
+                       )
+
+    df = df.drop([2,3,6,7,10,11,14,15,18,19]).reset_index(drop=True)
```
Comment (pissall20) [inline FI-Fully Insured/Dental-FI/Cigna-FI/enrollment-FI/code.py:31]: Column names and not index

File: FI-Fully Insured/Dental-FI/Sunlife-FI/code.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    df.to_csv(save_file_path, index=False)
+    print(f"Saved transformation for {file_path} to {save_file_path}")
+
+def clean_fi_dental_sunlife_confluent(file_path, save_file_path):
+    df = pd.read_excel(file_path,
+                       sheet_name="Summary Data", 
+                       skiprows=8, skipfooter=3,
+                       engine="openpyxl")
+    df = df.loc[:, ~df.columns.str.contains('^Unnamed')]
+    df=df.drop(19).reset_index(drop=True)
```
Comment (pissall20) [inline FI-Fully Insured/Dental-FI/Sunlife-FI/code.py:39]: Why drop 19? What's the logic behind that? What if other report has that kind of a value on a different row? Use logic and not just what you see and use the number of that row.

File: FI-Fully Insured/Medical-FI/Aetna-FI/claims-FI/code.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+
+
+def clean_fi_medical_aetna_claims(file_path, save_file_path):
+    df = pd.read_excel(file_path,
+                       sheet_name="BTM_12", 
+                       skiprows=14, skipfooter=14, header=[0,1],
+                       engine="openpyxl")
+    df.columns = df.columns.map('{0[0]} {0[1]}'.format)
+    cols=[0,1,5,7,9,11,12,13]
```
Comment (pissall20) [inline FI-Fully Insured/Medical-FI/Aetna-FI/claims-FI/code.py:28]: Use column names please

File: FI-Fully Insured/Medical-FI/Aetna-FI/claims-FI/code.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    df.drop(df.columns[cols],axis=1,inplace=True)
+    d={k: k.replace("Unnamed: 2_level_1", "") for k in df.columns}
+    df.rename(columns=d, inplace=True)
+    d1={k: k.replace("Unnamed: 3_level_1", "") for k in df.columns}
+    df.rename(columns=d1, inplace=True)
+    d2={k: k.replace("Unnamed: 4_level_1", "") for k in df.columns}
```
Comment (pissall20) [inline FI-Fully Insured/Medical-FI/Aetna-FI/claims-FI/code.py:34]: regex or chaining of replace commands

File: FI-Fully Insured/Medical-FI/Cigna-FI/california closets-FI/claims-FI/code.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+
+
+def clean_fi_medical_cigna_california_claims(file_path, save_file_path):
+    df = pd.read_excel(file_path,
+                       sheet_name="GC Exp R12-1",skiprows=8, skipfooter=15,
+                       engine="openpyxl"
+                       )
+
+    df = df.drop([12,13,15,16,23,24,37,38,51,52,65,66,79,80,87,88,101,102])
```
Comment (pissall20) [inline FI-Fully Insured/Medical-FI/Cigna-FI/california closets-FI/claims-FI/code.py:29]: Very very risky line of code when you drop so many columns based on numbers, you should use column names. What if the position of even 1 column changes? Your code will go for a toss.

File: FI-Fully Insured/Medical-FI/Cigna-FI/california closets-FI/large claim r12-FI/code.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+
+def clean_fi_medical_cigna_california_largeclaims_r12(file_path, save_file_path):
+    df = pd.read_excel(file_path,
+                       sheet_name="Sheet1",skiprows=8, skipfooter=2,
+                       engine="openpyxl"
+                       )
+
+    df=df.dropna()
+    df=df.drop(80).reset_index(drop=True)
```
Comment (pissall20) [inline FI-Fully Insured/Medical-FI/Cigna-FI/california closets-FI/large claim r12-FI/code.py:30]: Why drop 80? What's the logic behind it?

File: FI-Fully Insured/Medical-FI/Cigna-FI/california closets-FI/large claim ytd-FI/code.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+
+def clean_fi_medical_cigna_california_largeclaims_ytd(file_path, save_file_path):
+    df = pd.read_excel(file_path,
+                       sheet_name="Sheet1",skiprows=8, skipfooter=2,
+                       engine="openpyxl"
+                       )
+
+    df=df.dropna()
+    df=df.drop(20).reset_index(drop=True)
```
Comment (pissall20) [inline FI-Fully Insured/Medical-FI/Cigna-FI/california closets-FI/large claim ytd-FI/code.py:30]: Why drop 20? What's the logic behind it? What if that kind of row comes on a different row?

File: FI-Fully Insured/Medical-FI/Cigna-FI/fm industries-FI/claims-FI/code.py [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+
+
+def clean_fi_medical_cigna_fm_claims(file_path, save_file_path):
+    df = pd.read_excel(file_path,
+                       sheet_name="Sheet1",skiprows=8, skipfooter=2,
+                       engine="openpyxl"
+                       )
+
+    df=df.drop([6,7,14,15,22,23,30,31,38,39]).reset_index(drop=True)
```
Comment (pissall20) [inline FI-Fully Insured/Medical-FI/Cigna-FI/fm industries-FI/claims-FI/code.py:28]: Very very risky line of code when you drop so many columns based on numbers, you should use column names. What if the position of even 1 column changes? Your code will go for a toss.

File: FI-Fully Insured/Medical-FI/Cigna-FI/fm industries-FI/enrollment-FI/code.py [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+
+
+def clean_fi_medical_cigna_fm_enroll(file_path, save_file_path):
+    df = pd.read_excel(file_path,
+                       sheet_name="FI Enrollment by Grpd BenOpt-1", skiprows=5, skipfooter=8,
+                       engine="openpyxl"
+                       )
+
+    df=df.drop([5,6,12,13,19,20,26,27,33,34]).reset_index(drop=True)
```
Comment (pissall20) [inline FI-Fully Insured/Medical-FI/Cigna-FI/fm industries-FI/enrollment-FI/code.py:28]: Very very risky line of code when you drop so many columns based on numbers, you should use column names. What if the position of even 1 column changes? Your code will go for a toss.

File: FI-Fully Insured/Medical-FI/Cigna-FI/fm industries-FI/large claims-FI/code.py [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+
+def clean_fi_medical_cigna_fm_largeclaims(file_path, save_file_path):
+    df = pd.read_excel(file_path,
+                       sheet_name="Claims Exc De-ID Dx-1", skiprows=8, skipfooter=2,
+                       engine="openpyxl"
+                       )
+
+    df=df.dropna()
+    df=df.drop([10,16,24,30,34,38,42,50]).reset_index(drop=True)
```
Comment (pissall20) [inline FI-Fully Insured/Medical-FI/Cigna-FI/fm industries-FI/large claims-FI/code.py:29]: Very very risky line of code when you drop so many columns based on numbers, you should use column names. What if the position of even 1 column changes? Your code will go for a toss.

Print df.columns, then make a list of column names you want to drop and then use that list.

File: FI-Fully Insured/Medical-FI/Cigna-FI/paradigm-FI/claims-FI/code.py [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+
+
+def clean_fi_medical_cigna_paradigm_claims(file_path, save_file_path):
+    df = pd.read_excel(file_path,
+                       sheet_name="Exp by BenOpt-1", skiprows=6, skipfooter=7,
+                       engine="openpyxl"
+                       )
+
+    df=df.dropna(axis=0, subset=['BENEFIT OPTION'])
+    df=df.drop([3,9,15,20,25,])
```
Comment (pissall20) [inline FI-Fully Insured/Medical-FI/Cigna-FI/paradigm-FI/claims-FI/code.py:29]: Same as above

File: FI-Fully Insured/Medical-FI/Cigna-FI/paradigm-FI/enrollment-FI/code.py [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+
+def clean_fi_medical_cigna_paradigm_enroll(file_path, save_file_path):
+    df = pd.read_excel(file_path,
+                       sheet_name="Enrollment by Tier-1", skiprows=4, skipfooter=18,
+                       engine="openpyxl"
+                       )
+
+    df=df.dropna(axis=0, subset=['BENEFIT OPTION'])
+    df=df.drop([14,30,46,62,78,])
```
Comment (pissall20) [inline FI-Fully Insured/Medical-FI/Cigna-FI/paradigm-FI/enrollment-FI/code.py:29]: Same as above

File: LF-Level Funded/Medical-LF/Premera-LF/claims-LF/code.py [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+
+
+
+def clean_lf_premera_claims(file_path, save_file_path):
+    df = pd.read_excel(file_path,
+                       sheet_name="4020791-SANA BIOTECHNOLOGY INC|", 
+                       skiprows=6, skipfooter=2, header=[0,1],
+                       engine="openpyxl") 
+    df.columns = df.columns.map('{0[0]} {0[1]}'.format)
+    cols=[1,3,4,6,8,10,11,12,14,15,16,18]
```
Comment (pissall20) [inline LF-Level Funded/Medical-LF/Premera-LF/claims-LF/code.py:28]: Very very risky line of code when you drop so many columns based on numbers, you should use column names. What if the position of even 1 column changes? Your code will go for a toss.

File: LF-Level Funded/Medical-LF/Premera-LF/enrollment-LF/code.py [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+def create_save_file_path(read_file_path, tran_suffix="_T"):
+    actual_path = read_file_path.split(".")[0]
+    save_file_path = actual_path + tran_suffix + ".csv"
+    return save_file_path
+
+
+
+def clean_lf_premera_enrolment(file_path, save_file_path):
+    df = pd.read_excel(file_path,
+                       sheet_name="M|4020791-SANA BIOTECHNOLOGY INC",
```
Comment (pissall20) [inline LF-Level Funded/Medical-LF/Premera-LF/enrollment-LF/code.py:24]: Is this an excel with a single sheet? What if the sheet name comes different if there is a logic to that number? If the excel has only 1 sheet, then don't pass sheet_name

File: LF-Level Funded/Medical-LF/Premera-LF/largeclaims-LF/code.py [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+def create_save_file_path(read_file_path, tran_suffix="_T"):
+    actual_path = read_file_path.split(".")[0]
+    save_file_path = actual_path + tran_suffix + ".csv"
+    return save_file_path
+
+
+
+def clean_lf_premera_large_claims(file_path, save_file_path):
+    df = pd.read_excel(file_path,
+                       sheet_name="4020791-SANA BIOTECHNOLOGY INC|",
```
Comment (pissall20) [inline LF-Level Funded/Medical-LF/Premera-LF/largeclaims-LF/code.py:24]: Is this an excel with a single sheet? What if the sheet name comes different if there is a logic to that number? If the excel has only 1 sheet, then don't pass sheet_name

File: FI-Fully Insured/Dental-FI/Cigna-FI/claims-FI/code.py [hunk 17]
```diff
@@ -0,0 +1,130 @@
...
+                       sheet_name="Acct_Product Grp-1", skiprows=6, skipfooter=5,
+                       engine="openpyxl"
+                       )
+
+    df = df[(df['PRODUCT GROUP'] == 'DENT')].reset_index(drop=True)
+    df.to_csv(save_file_path, index=False)
+    print(f"Saved transformation for {file_path} to {save_file_path}")
+
+
+def clean_fi_dental_cigna_claims_financialforce(file_path, save_file_path):
```
Comment (pissall20) [inline FI-Fully Insured/Dental-FI/Cigna-FI/claims-FI/code.py:46]: These reports are all similar, they definitely don't require so many functions to transform them. Please review your logic and correct it.

File: FI-Fully Insured/Dental-FI/Cigna-FI/enrollment-FI/code.py (hunk 2 above)
Comment (gdawg32) [reply FI-Fully Insured/Dental-FI/Cigna-FI/enrollment-FI/code.py:31]: I was dropping rows here. Is it alright to keep it?

Comment (gdawg32) [reply FI-Fully Insured/Dental-FI/Cigna-FI/enrollment-FI/code.py:31]: Ok i figured to drop without index

File: LF-Level Funded/Medical-LF/Premera-LF/enrollment-LF/code.py (hunk 15 above)
Comment (gdawg32) [reply LF-Level Funded/Medical-LF/Premera-LF/enrollment-LF/code.py:24]: There are multiple sheets, so this is the only way to get the first sheet

File: LF-Level Funded/Medical-LF/Premera-LF/largeclaims-LF/code.py (hunk 16 above)
Comment (gdawg32) [reply LF-Level Funded/Medical-LF/Premera-LF/largeclaims-LF/code.py:24]: Multiple sheet it got

**Pull request merged**